### DISCUSSION TITLE
Communication/Tracking with SmartSolar MPPT

### DISCUSSION THREAD

**Discussion by rychom** [hoylabs/OpenDTU-OnBattery#1003] [Q&A]
Hello everyone,

I've already read a lot about how the individual components are assembled.
I still have some gaps in my knowledge about communication and reading out the SmartSolar MPPT.
The questions are also touched on here in the Wiki, but they don't take me any further.
What is currently the simplest and cheapest way to read out the SmartSolar MPPT from Victron in 2024? I have an openDTU-onBattery, but it is in the house and the MPPT is placed on the balcony.
As I understand it now, it is not necessary for the openDTU to communicate with the MPPT, but there are disadvantages:
- The battery may be charged/discharged excessively (is this true?)
- Battery status can only be read out via Bluetooth and cannot be tracked.

The best thing for my setup would be a web API for the MPPT. Can something like this be realized easily and cheaply?

Further information on the current (planned) setup:
- Hitchi IR on the electricity meter
- openDTU-onBattery
- 2x Winaico 410Wp
- HM-800
- 24V/50Ah or 24V/50Ah battery
- MPPT SmartSolar 100/20

Thank you very much!

**Comment by schlimmchen** [comment]
> I have an openDTU-onBattery, but it is in the house and the MPPT is placed on the balcony.

And you cannot run a cable between the two? Then you are out of luck regarding solar passthrough.

> The battery may be charged/discharged excessively

Why should this be the case?

> Battery status can only be read out via Bluetooth and cannot be tracked.

Bluetooth? There is no support to connect any peripheral using Bluetooth.

> The best thing for my setup would be a web API for the MPPT.

That is a popular request. I don't see how that helps you... Maybe I am misunderstanding? Who serves the web API, what is served, and who uses the web API?

> Can something like this be realized easily and cheaply?

Reading the solar output from a web API or MQTT topic is the popular request that I mentioned. It would allow to use solar-passthrough without VE.Direct, but a different data source. Such an implementation needs significant work.

In summary: If you cannot run a cable between your OpenDTU and the Victron charge controller, you cannot use solar-passthrough. Instead, the battery will be charged until it reaches the start threshold. Until then, the inverter will be off. After the start threshold is reached, the battery may be discharged, and the inverter is enabled to match the household consumption.

**Reply by rychom** [reply]
Thank you for the fast reply!

> Bluetooth? There is no support to connect any peripheral using Bluetooth.

Ah maybe I got this wrong. I thought you can see the Battery Status via Victron APP when you're connected via BT.

> That is a popular request. I don't see how that helps you... Maybe I am misunderstanding? Who serves the web API, what is served, and who uses the web API?

I would like to use the web API to communicate with the MPPT instead of the VE.Direct link. So my basic question is: Whats the best, simplest and cheapest way to let them communicate without using VE.Direct?

> Reading the solar output from a web API or MQTT topic is the popular request that I mentioned. It would allow to use solar-passthrough without VE.Direct, but a different data source. Such an implementation needs significant work.
> 
> In summary: If you cannot run a cable between your OpenDTU and the Victron charge controller, you cannot use solar-passthrough. Instead, the battery will be charged until it reaches the start threshold. Until then, the inverter will be off. After the start threshold is reached, the battery may be discharged, and the inverter is enabled to match the household consumption.

I am a bit lost here. So you can implement solar-passthrough without VE.Direct with significant work, but you cannot use solar-passthrough when the DTU and MPPT are not connected? Isnt the connection between them VE.Direct? Sorry if i misunderstand that.

Maybe i have some deeper lacks of knowlede. What I am thinking (without DTU, battery has own unknown BMS, inverter between battery and house):
![mppt_batt](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/24256351/46f34578-19a5-47c3-a79b-9c640f23de3e)
My inverter limited to e.g. 200W.
PV generates 400W
So the MPPT is passing this 400W towards Battery&inverter. As the inverter takes 200W, the battery will take the other 200W?
Or is the battery taking everything and the inverter wont start? But is it easier to load a battery then to invert it? im confused..

**Comment by Manos1966** [comment]
> But is it easier to load a battery then to invert it? im confused..

Electricity takes always the "way of minimal resistance".
This is the reason why electricity will prefer to pass through your body (instead of the N cable) when you touch 230V with your hand...

Coming back to our 48V now, the battery has higher resistance than the inverter. As such, the electricity will always chose to go to the inverter first.
Subesquently,  the MPPT is passing this 400W towards Battery&inverter. The inverter takes 200W, the battery will take the other 200W.
